apps/examples/testcase/ta_tc/wifi_manager/utc/utc_wifi_manager_main.c

```diff
@@ -0,0 +1,348 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <pthread.h>
+#include <sys/time.h>
+#include <errno.h>
+#include <apps/shell/tash.h>
+#include <wifi_manager/wifi_manager.h>
+#include "../../../tc_common.h"
+
+extern sem_t tc_sem;
+extern int working_tc;
+
+static void wifi_sta_connected(void);		// in station mode, connected to ap
+static void wifi_sta_disconnected(void);	// in station mode, disconnected from ap
+static void wifi_softap_sta_joined(void);	// in softap mode, a station joined
+static void wifi_softap_sta_left(void);		// in softap mode, a station left
+static void wifi_sta_scan_ap_done(wifi_manager_scan_info_s **scan_info, wifi_manager_scan_result_e res); // in station mode, scanning ap is done
+
+static wifi_manager_cb_s wifi_callbacks = {
+	NULL,	//wifi_sta_connected,
+	NULL,	//wifi_sta_disconnected,
+	NULL,	//wifi_softap_sta_joined,
+	NULL,	//wifi_softap_sta_left,
+	wifi_sta_scan_ap_done,	// in station mode, this callback function is called when scanning ap is done.
+};
+
+static wifi_manager_cb_s wifi_null_callbacks = {
+	NULL,	//wifi_sta_connected,
+	NULL,	//wifi_sta_disconnected,
+	NULL,	//wifi_softap_sta_joined,
+	NULL,	//wifi_softap_sta_left,
+	NULL,	// in station mode, this callback function is called when scanning ap is done.
+};
+
+static void wifi_sta_scan_ap_done(wifi_manager_scan_info_s **scan_info, wifi_manager_scan_result_e res)
+{
+	printf("[pkes] T%d --> %s\n", getpid(), __FUNCTION__);
+	/* Make sure you copy the scan results onto a local data structure.
+	 * It will be deleted soon eventually as you exit this function.
+	 */
+	if (res == WIFI_SCAN_FAIL) {
+		printf("WiFi scan failed\n");
+		return;
+	}
+	wifi_manager_scan_info_s *wifi_scan_iter = *scan_info;
+	while (wifi_scan_iter != NULL) {
+		printf("WiFi AP SSID: %-20s, WiFi AP BSSID: %-20s, WiFi Rssi: %d\n", wifi_scan_iter->ssid, wifi_scan_iter->bssid, wifi_scan_iter->rssi);
+		wifi_scan_iter = wifi_scan_iter->next;
+	}
+}
+
+static void utc_wifi_manager_init_n(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	ret = wifi_manager_init(NULL);
+
+	TC_ASSERT_EQ("wifi_manager_init", ret, WIFI_MANAGER_FAIL);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_wifi_manager_init_p(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	ret = wifi_manager_init(&wifi_callbacks);
+
+	TC_ASSERT_EQ("wifi_manager_init", ret, WIFI_MANAGER_SUCCESS);
+	TC_SUCCESS_RESULT();
+}
+
+static void utc_wifi_manager_set_mode_n(void)
+{
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	ret = wifi_manager_set_mode(SOFTAP_MODE, NULL);
+
+	TC_ASSERT_EQ("wifi_manager_set_mode", ret, WIFI_MANAGER_FAIL);
+
+	wifi_manager_softap_config_s ap_config = {"PKES", 1, "1111122222"};
```

Comment:
how about to set AP's configurations by Kconfig

Reply:
I do change.